### PR TITLE
Allow xml docs to still be created when 'emit metadata only' is on.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
EmitMetadataOnly is a flag that exists for IDE scenarios to allow it to emit dlls cheaply that give hte top-level surface area of a particular compilation.  This can then be used in cross-language scenarios to allow things like intellisense/QI/and-lots-more to work in a live fashion.  We call these cross-language metadata-only dlls: 'skeleton assemblies'.

The IDE would like to improve our story around skeleton assemblies by being able to cache them to our persistence store so that later launched sessions of VS can reuse them without recomputing them.  

This is generally not difficult for us to do except for one small snag.  Specifically, one thing we support *today* with skeletons is being able to expose xml doc comments through the skeleton.  This works by having us actually both emit the metadta-only-dll, but also creating a DocumentationProvider that wraps the *original language* compilation.  So, for example, if VB references C#, it will get a dll for the C# metadata, but it will get a doc-provider that forwards along to the C# compilation to answer questions.

If we want to persist the dll (and still have xml-doc-comments work across languages) then we need to persist the xml doc info as well.  This PR updates the  compiler to support that.  Previously, the compiler would completely skip xml doc generation if EmitMetadataOnly was set.  With this PR, the compiler will emit xml doc comments as well if the appropriate stream is provided.

Relates to ref assemblies: https://github.com/dotnet/roslyn/issues/18612

### CODE REVIEW

File: src/Compilers/Core/Portable/Compilation/Compilation.cs
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/Compilation/Compilation.cs:2670]:
```suggestion
                        // If we're in metadata only, and the caller asks for xml docs, then still proceed and generate those.
```

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler  ptal :)

File: src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs:3313]: Is this parameter used?

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs:280]: Consider verifying that "This should be emitted" made it into the output. `Length > 0` is not a strong test.

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs:249]: What does "NoDocMode" mean?

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 7)

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs:249]: SureSure!

File: src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs:3313]: nope!  removing. thanks :)

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs:249]: done.

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs:260]: >ConEmitMetadataOnly_XmlDocs_NoDocMode_Successsole [](http://example.com/codeflow?start=12&length=49)

Accidental paste?

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 10) modulo typo in test

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for second review. Thanks

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs:485]: nit: Please remove this comment here and the one below `// Even though docs failed, ...`

**Review by RikkiGibson — APPROVED** [review]
LGTM except the feedback above (just have to address before merge).

**Comment by jcouv** [comment]
Feel free to merge/squash once the typo and last comment are addressed/considered. Thanks

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs:260]:
```suggestion
            EmitMetadataOnly_XmlDocs_NoDocMode_Success.WriteLine(""hello"");
```

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs:485]:
```suggestion
```

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs:488]:
```suggestion
```

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs]: This doesn't look right, aren't we trying to call Console.WriteLine here?

Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs]: sigh...

File: src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs:260]:
```suggestion
            Console.WriteLine(""hello"");
```

**Pull request merged**
